Lib/test/test_socket.py

```diff
@@ -1259,6 +1259,38 @@ def testGetServBy(self):
         # Find one service that exists, then check all the related interfaces.
         # I've ordered this by protocols that have both a tcp and udp
         # protocol, at least for modern Linuxes.
+        def query_available_service(*protocols):
```

Comment:
```suggestion
        def find_service(expected_protocols):
```

Make it a real function and make it accept a list of expected protocols as well as a default services_file set to `/etc/services`.